wiseconnect/components/device/silabs/si91x/mcu/drivers/unified_api/src/sl_si91x_driver_gpio.c

```diff
@@ -783,7 +783,8 @@ sl_status_t sl_si91x_gpio_driver_enable_pad_selection(uint8_t gpio_padnum)
 sl_status_t sl_si91x_gpio_driver_enable_host_pad_selection(uint8_t gpio_num)
 {
   // Check if the GPIO pin number exceeds the maximum allowed value.
-  if (gpio_num >= HOST_PAD_MIN && gpio_num <= HOST_PAD_MAX) {
+  //if (gpio_num >= HOST_PAD_MIN && gpio_num <= HOST_PAD_MAX) { FIXME SDK bug
+  if (gpio_num < HOST_PAD_MIN || gpio_num > HOST_PAD_MAX) {
```

Comment:
This is fixed in a different way in Wiseconnect 3.3.1. It would be cleaner if we apply the exact same change as upstream for now:

```
  if (!(gpio_num >= HOST_PAD_MIN && gpio_num <= HOST_PAD_MAX)) {
```

Also, patches should preferably be done in separate commits from the ones introducing unmodified code from upstream.

We also want to upgrade to 3.3.2 soon, but let's keep that separate from this PR.

Reply:
I did not want to add this to this PR, will revert.